File: examples/python/news.py
 ---
  import requests

  API_KEY = "your_api_key_here"
  BASE_URL = "https://nestdaddy.com/api/v1"
  HEADERS = {"X-API-Key": API_KEY}


  def search(query, limit=10):
      r = requests.get(
          f"{BASE_URL}/search",
          headers=HEADERS,
          params={"q": query, "limit": limit}
      )
      r.raise_for_status()
      return r.json()


  def get_tech_news(category="all", limit=10):
      r = requests.get(
          f"{BASE_URL}/news/tech",
          headers=HEADERS,
          params={"category": category, "limit": limit}
      )
      r.raise_for_status()
      return r.json()


  def get_financial_news(category="all", limit=10):
      r = requests.get(
          f"{BASE_URL}/news/financial",
          headers=HEADERS,
          params={"category": category, "limit": limit}
      )
      r.raise_for_status()
      return r.json()


  def get_global_news(limit=10):
      r = requests.get(
          f"{BASE_URL}/news/global",
          headers=HEADERS,
          params={"limit": limit}
      )
      r.raise_for_status()
      return r.json()


  if __name__ == "__main__":
      print("=== Web Search ===")
      results = search("artificial intelligence")
      for item in results["results"]:
          print(f"  {item['title']}")
          print(f"  {item['url']}\n")

      print("=== Tech News (AI) ===")
      news = get_tech_news(category="ai")
      for article in news["articles"]:
          print(f"  {article['title']}")
          print(f"  {article['published_at']}\n")

      print("=== Financial News (Crypto) ===")
      news = get_financial_news(category="crypto")
      for article in news["articles"]:
          print(f"  {article['title']}\n")

  ---

File: examples/javascript/news.js
 const API_KEY = "your_api_key_here";                                                                                                                                                                               const BASE_URL = "https://nestdaddy.com/api/v1";                                                                                                                                                                   const HEADERS = { "X-API-Key": API_KEY };

  async function search(query, limit = 10) {
    const res = await fetch(
      `${BASE_URL}/search?q=${encodeURIComponent(query)}&limit=${limit}`,
      { headers: HEADERS }
    );
    if (!res.ok) throw new Error(`HTTP ${res.status}`);
    return res.json();
  }

  async function getTechNews(category = "all", limit = 10) {
    const res = await fetch(
      `${BASE_URL}/news/tech?category=${category}&limit=${limit}`,
      { headers: HEADERS }
    );
    if (!res.ok) throw new Error(`HTTP ${res.status}`);
    return res.json();
  }

  async function getFinancialNews(category = "all", limit = 10) {
    const res = await fetch(
      `${BASE_URL}/news/financial?category=${category}&limit=${limit}`,
      { headers: HEADERS }
    );
    if (!res.ok) throw new Error(`HTTP ${res.status}`);
    return res.json();
  }

  async function getGlobalNews(limit = 10) {
    const res = await fetch(
      `${BASE_URL}/news/global?limit=${limit}`,
      { headers: HEADERS }
    );
    if (!res.ok) throw new Error(`HTTP ${res.status}`);
    return res.json();
  }

  // Example usage
  (async () => {
    console.log("=== Web Search ===");
    const results = await search("artificial intelligence");
    results.results.forEach(r => console.log(`  ${r.title}\n  ${r.url}\n`));

    console.log("=== Tech News (AI) ===");
    const techNews = await getTechNews("ai");
    techNews.articles.forEach(a => console.log(`  ${a.title}\n  ${a.published_at}\n`));

    console.log("=== Financial News (Crypto) ===");
    const finNews = await getFinancialNews("crypto");
    finNews.articles.forEach(a => console.log(`  ${a.title}\n`));
  })();

  ---
